docs/agent-llm-workflow.md

```
@@ -0,0 +1,120 @@
+            },
+            "projectRoot": "/path/to/project" // Agent should provide this
+        }
+        ```
+
+5.  **Taskmaster Processes Agent's Response & Resumes Original Operation:**
+    *   Taskmaster's `agent_llm` tool receives the call from the agent.
+    *   The `TaskMasterMCPServer` core logic uses the `interactionId` to find the paused operation.
+    *   If `agentLLMResponse.status` is `"success"`, the `agentLLMResponse.data` is used as the result of the originally delegated LLM call (e.g., as if `generateText` returned this data). The paused operation resumes and completes.
+    *   If `agentLLMResponse.status` is `"error"`, the `agentLLMResponse.errorDetails` are used to signal an error in the paused operation. The operation typically fails or retries based on Taskmaster's internal error handling.
+    *   The `agent_llm` tool itself will respond to the agent's call with a confirmation like:
+    
```

markdownlint-cli2 (0.17.2)

Trailing spaces
Expected: 0 or 2; Actual: 4

(MD009, no-trailing-spaces)